tests/testthat/test-evaluate.r

```diff
@@ -47,6 +47,20 @@ test_that("options(warn = -1) suppresses warnings", {
   expect_that(classes(ev), equals("source"))
 })
 
+test_that("options(warn = 0) and options(warn = 1) produces warnings", {
+  ev <- evaluate("op = options(warn = 0); warning('hi'); options(op)")
+  expect_equal(classes(ev), c("source", "simpleWarning"))
+
+  ev <- evaluate("op = options(warn = 1); warning('hi'); options(op)")
+  expect_equal(classes(ev), c("source", "simpleWarning"))
+})
+
+# errors won't work regularly in test_that()
+ev_warn_2 <- evaluate("op = options(warn = 2); warning('hi'); options(op)")
+test_that("options(warn = 2) produces errors instead of warnings", {
+  expect_equal(classes(ev_warn_2), c("source", "simpleError"))
```

Comment:
@yutannihilation I don't know why, but this test actually fails. Could you investigate it? Sorry I just realized it.

Reply:
@yihui Sure! Will take a look on this weekend.

Reply:
Considering the fact
https://travis-ci.org/r-lib/evaluate/builds/267994531 succeeded and https://travis-ci.org/r-lib/evaluate/builds/268000616 failed, this seems due to some change introduced in devtools 1.13.4 or 1.13.5.

Reply:
> this seems due to some change introduced in devtools 1.13.4 or 1.13.5.

This assumption was wrong. The cause seems the change in testthat 2.0.0. I've confirmed the test passes w/ testthat 1.0.2.